src/project/router.js

```diff
@@ -0,0 +1,136 @@
+const { Router } = require("express");
+const auth = require("../auth/middleWare");
+const Project = require("./model");
+const Studio = require("../studio/model");
+const Image = require("../image/model");
+const router = new Router();
+
+router.get("/project", async function(request, response, next) {
+  try {
+    const projects = await Project.findAll();
+    response.send(projects);
+    console.log("done");
+  } catch (error) {
+    next(error);
+  }
+});
+
+router.get("/project/:id", async function(request, response, next) {
+  try {
+    const project = await Project.findByPk(request.params.id);
+    response.send(project);
+    console.log("done");
+  } catch (error) {
+    next(error);
+  }
+});
+
+router.post("/studio/:id/project", auth, async function(
+  request,
+  response,
+  next
+) {
+  try {
+    console.log("how my request looks?", request.user.dataValues.id);
+    console.log("WHAT IS STUDIO ID ", request.params.id);
+
+    const studio = await Studio.findByPk(request.params.id);
+    if (studio) {
+      request.body.projectDetails.userId = request.user.dataValues.id;
+      console.log("HOW MY REQUEST  BODY LOOOKS ", request.body.projectDetails);
+      const newProject = { ...request.body };
+      console.log("what is my new !!!!! project ", newProject.projectDetails);
+      newProject.projectDetails.studioId = request.params.id;
```

Comment:
here you are mutating request.body and later new project, this is not recommended, a better way of doing this would be:
```
const newProject = { ...request.body, projectDetails: { ...request.body.projectDetails, userId: request.user.dataValues.id,  studioId: request.params.id } }
```
nice job eitherway playing around with these objects to get what you want! 😄